cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -360,6 +361,15 @@ class SnappyUnifiedMemoryManager private[memory](
           return
         }
 
+        // Stop execution pool to grow beyond eviction percentage of heap.
+        if (memoryMode eq MemoryMode.ON_HEAP) {
+          if (executionPool.memoryUsed + extraMemoryNeeded > maxHeapExecutionSize) {
+            logWarning(s"MemoryManager can't allocate $numBytes bytes as it " +
+                s"would exceed maxHeapExecutionSize = $maxHeapExecutionSize")
+            return
+          }
+        }
+
```

Comment:
Hmm, then this needs to be re-looked at. We should assume that in a running cluster there are concurrent queries going on which can keep the execution area occupied continuously. If user wants to ingest something into storage then it will be trouble in terms of memory use. We should change this so that execution never grows into storage as per the fraction for both heap and off-heap. At least lower the max limit of execution to be much smaller, and closer to fraction rather than eviction limit.